bitnami/cassandra/templates/statefulset.yaml

```diff
@@ -68,12 +68,29 @@ spec:
           image: {{ include "cassandra.volumePermissions.image" . }}
           imagePullPolicy: {{ .Values.volumePermissions.image.pullPolicy | quote }}
           command:
-            - /bin/bash
-            - -ec
+            - /bin/sh
+            - -cx
             - |
-              chown -R {{ .Values.containerSecurityContext.runAsUser }}:{{ .Values.podSecurityContext.fsGroup }} /bitnami/cassandra
-          securityContext:
-            runAsUser: 0
+              {{- if .Values.persistence.enabled }}
+              {{- if eq ( toString ( .Values.volumePermissions.securityContext.runAsUser )) "auto" }}
+              chown `id -u`:`id -G | cut -d " " -f2` /bitnami/cassandra
+              {{- else }}
+              chown {{ .Values.containerSecurityContext.runAsUser }}:{{ .Values.podSecurityContext.fsGroup }} /bitnami/cassandra
+              {{- end }}
+              mkdir -p /bitnami/cassandra/data
+              chmod 700 /bitnami/cassandra/data
+              find /bitnami/cassandra -mindepth 1 -maxdepth 1 -not -name ".snapshot" -not -name "lost+found" | \
+              {{- if eq ( toString ( .Values.volumePermissions.securityContext.runAsUser )) "auto" }}
+                xargs chown -R `id -u`:`id -G | cut -d " " -f2`
+              {{- else }}
+                xargs chown -R {{ .Values.containerSecurityContext.runAsUser }}:{{ .Values.podSecurityContext.fsGroup }}
+              {{- end }}
```

Comment:
I'm not sure if this part is needed for Cassandra or it is a leftover when copying the PostgreSQL code block

Reply:
It should be needed to set the owner to the mounted /secrets or /conf dir.